tests/phpunit/integration/Modules/AnalyticsTest.php

```diff
@@ -259,6 +259,46 @@ public function tracking_disabled_provider() {
 		);
 	}
 
+	/**
+	 * @dataProvider data_determine_account_id
+	 */
+	public function test_determine_account_id( $property_id, $expected ) {
+		$class  = new \ReflectionClass( Analytics::class );
+		$method = $class->getMethod( 'determine_account_id' );
+		$method->setAccessible( true );
```

Comment:
I'm not sure if we should add unit tests for protected methods in this way. Ideally, we would be able to add coverage for protected methods implicitly rather than by invoking them directly from outside. 

Obviously it is easily testable by itself as it's a pure function, so I agree that this is useful but I think it somewhat indicates that maybe it shouldn't be a protected method. Is there any reason why we wouldn't add an actual function for something like this instead? Maybe some functions would live under a service-specific namespace rather than related to the module? E.g. `Google\Site_Kit\Service\Analytics\`

Now that our minimum version of PHP is 5.6, we can [use/import functions](https://www.php.net/manual/en/language.namespaces.importing.php) so having a few of our own would be just as clean to implement.

Reply:
I agree that's not great, but let's keep this protected for now. The only reason we'd change it is for testability at this point, but that's still possible via the "hack" here. We can re-evaluate in the future if needed.